lib/core/utils/filter-html-attrs.js

```diff
@@ -0,0 +1,178 @@
+  `(?:"([^"]*)"` + // group 2: double quoted attribute value; look for everything surrounded by double quotes
+  `|'([^']*)'` + // group 3: single quoted attribute value; look for everything surrounded by single quotes
+  `|([^'"\\s]*)))` + // group 4: non-quoted attribute value; look for any non-whitespace, non-quote characters
+    `|\\b(\\S+)(?:\\B|\\b)`, // group 5: non-valued attribute name; look for any remaining non-matched strings surrounded by word boundaries
+  'g'
+);
+
+const attrMap = {
+  for: 'htmlFor',
+  class: 'className'
+};
+
+/**
+ * Match an attribute to a matcher object.
+ * @param {String} nodeName - Name of the current node
+ * @param {Object[]} attributes - Array of node attributes and values
+ * @param {String} value - Value of the current attribute
+ * @param {any} matcher - How to match the value
+ */
+function matches(nodeName, attributes, value, matcher) {
+  switch (typeof matcher) {
+    case 'boolean':
+      return true;
+    case 'string':
+      return matcher === value;
+    case 'object':
+      if (Array.isArray(matcher)) {
+        return matcher.includes(value);
+      }
+
+      // matcher-like object, match all parts
+      let filter = true;
+
+      // match each property
+      if (matcher.nodeName) {
+        filter = filter && matcher.nodeName === nodeName;
+      }
+
+      if (matcher.attributes) {
+        Object.keys(matcher.attributes).forEach(attrName => {
+          const attr = attributes.find(attr => attr.name === attrName);
+          if (!attr) {
+            return false;
+          }
+
+          filter =
+            filter &&
+            matches(
+              nodeName,
+              attributes,
+              attr.value,
+              matcher.attributes[attrName]
+            );
+        });
+      }
+
+      return filter;
+  }
+}
+
+/**
+ * Parse an html string for each tag
+ * @param {String} htmlString
+ * @returns {Object[]} Parsed tags with nodeName, attributes, and raw properties
+ */
+function parseElements(htmlString) {
+  const elements = [];
+  let match;
+  while ((match = tagStringRegex.exec(htmlString))) {
+    const tagStr = match[1];
+
+    // no need to filter end tags
+    if (tagStr.charAt(0) === '/') {
+      continue;
+    }
+
+    const index =
+      tagStr.indexOf(' ') !== -1 ? tagStr.indexOf(' ') : tagStr.length;
+    const nodeName = tagStr.substring(0, index).toLowerCase();
+    const attributeStr = tagStr.substring(index + 1).trim();
+
+    let groups;
+    const attributes = [];
+    while ((groups = attributeRegex.exec(attributeStr))) {
+      attributes.push({
+        // can be either a valued and non-valued attribute
+        name: (groups[1] || groups[5]).toLowerCase(),
+        // double, single, or non-quoted value
+        value: groups[2] || groups[3] || groups[4] || '',
+        raw: groups[0]
+      });
+    }
```

Comment:
This seems like it should be:

```suggestion
    const attributes = parseAttributes(attributeStr);
```